runtime/src/main/java/org/corfudb/runtime/object/MVOCorfuCompileProxy.java

```diff
@@ -0,0 +1,222 @@
+package org.corfudb.runtime.object;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.common.metrics.micrometer.MicroMeterUtils;
+import org.corfudb.protocols.logprotocol.SMREntry;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+import org.corfudb.protocols.wireprotocol.TxResolutionInfo;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.AbortCause;
+import org.corfudb.runtime.exceptions.NetworkException;
+import org.corfudb.runtime.exceptions.TransactionAbortedException;
+import org.corfudb.runtime.object.transactions.AbstractTransactionalContext;
+import org.corfudb.runtime.object.transactions.TransactionalContext;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.util.ReflectionUtils;
+import org.corfudb.util.serializer.ISerializer;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@Slf4j
+public class MVOCorfuCompileProxy<T extends ICorfuSMR<T>> implements ICorfuSMRProxyInternal<T> {
+
+    @Getter
+    MultiVersionObject<T> underlyingMVO;
+
+    final CorfuRuntime rt;
+
+    final UUID streamID;
+
+    final Class<T> type;
+
+    @Getter
+    ISerializer serializer;
+
+    @Getter
+    Set<UUID> streamTags;
+
+    private final Object[] args;
+
+    public MVOCorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[] args,
+                                ISerializer serializer, Set<UUID> streamTags, ICorfuSMR<T> wrapperObject) {
+        this.rt = rt;
+        this.streamID = streamID;
+        this.type = type;
+        this.args = args;
+        this.serializer = serializer;
+        this.streamTags = streamTags;
+
+        this.underlyingMVO = new MultiVersionObject<>(
+                rt,
+                this::getNewInstance,
+                new StreamViewSMRAdapter(rt, rt.getStreamsView().getUnsafe(streamID)),
+                wrapperObject, streamID);
```

Comment:
This is a memory leak. Since it is not a  VersionLockedObject the internal data structures that are populated inside the StreamView are never garbage collected via `ViewsGarbageCollector`